from PyQt5 import QtCore, QtGui, QtWidgets


class Ui_eightPuzzle(object):
    def setupUi(self, eightPuzzle):
        eightPuzzle.setObjectName("eightPuzzle")
        eightPuzzle.resize(939, 408)
        font = QtGui.QFont()
        font.setFamily("微软雅黑")
        eightPuzzle.setFont(font)
        self.init6 = QtWidgets.QLineEdit(eightPuzzle)
        self.init6.setGeometry(QtCore.QRect(130, 100, 31, 31))
        self.init6.setAlignment(QtCore.Qt.AlignCenter)
        self.init6.setObjectName("init6")
        self.final2 = QtWidgets.QLineEdit(eightPuzzle)
        self.final2.setGeometry(QtCore.QRect(240, 60, 31, 31))
        self.final2.setAlignment(QtCore.Qt.AlignCenter)
        self.final2.setObjectName("final2")
        self.init3 = QtWidgets.QLineEdit(eightPuzzle)
        self.init3.setGeometry(QtCore.QRect(130, 60, 31, 31))
        self.init3.setAlignment(QtCore.Qt.AlignCenter)
        self.init3.setObjectName("init3")
        self.final7 = QtWidgets.QLineEdit(eightPuzzle)
        self.final7.setGeometry(QtCore.QRect(200, 140, 31, 31))
        self.final7.setAlignment(QtCore.Qt.AlignCenter)
        self.final7.setObjectName("final7")
        self.show1 = QtWidgets.QLabel(eightPuzzle)
        self.show1.setGeometry(QtCore.QRect(580, 50, 50, 50))
        font = QtGui.QFont()
        font.setFamily("Geeza Pro")
        font.setPointSize(14)
        self.show1.setFont(font)
        self.show1.setLayoutDirection(QtCore.Qt.LeftToRight)
        self.show1.setAlignment(QtCore.Qt.AlignCenter)
        self.show1.setObjectName("show1")
        self.final9 = QtWidgets.QLineEdit(eightPuzzle)
        self.final9.setGeometry(QtCore.QRect(280, 140, 31, 31))
        self.final9.setAlignment(QtCore.Qt.AlignCenter)
        self.final9.setObjectName("final9")
        self.show5 = QtWidgets.QLabel(eightPuzzle)
        self.show5.setGeometry(QtCore.QRect(630, 100, 50, 50))
        font = QtGui.QFont()
        font.setFamily("Geeza Pro")
        font.setPointSize(14)
        self.show5.setFont(font)
        self.show5.setAlignment(QtCore.Qt.AlignCenter)
        self.show5.setObjectName("show5")
        self.init9 = QtWidgets.QLineEdit(eightPuzzle)
        self.init9.setGeometry(QtCore.QRect(130, 140, 31, 31))
        self.init9.setAlignment(QtCore.Qt.AlignCenter)
        self.init9.setObjectName("init9")
        self.init7 = QtWidgets.QLineEdit(eightPuzzle)
        self.init7.setGeometry(QtCore.QRect(50, 140, 31, 31))
        self.init7.setAlignment(QtCore.Qt.AlignCenter)
        self.init7.setObjectName("init7")
        self.label_10 = QtWidgets.QLabel(eightPuzzle)
        self.label_10.setGeometry(QtCore.QRect(220, 10, 71, 31))
        self.label_10.setAlignment(QtCore.Qt.AlignCenter)
        self.label_10.setObjectName("label_10")
        self.init5 = QtWidgets.QLineEdit(eightPuzzle)
        self.init5.setGeometry(QtCore.QRect(90, 100, 31, 31))
        self.init5.setAlignment(QtCore.Qt.AlignCenter)
        self.init5.setObjectName("init5")
        self.show2 = QtWidgets.QLabel(eightPuzzle)
        self.show2.setGeometry(QtCore.QRect(630, 50, 50, 50))
        font = QtGui.QFont()
        font.setFamily("Geeza Pro")
        font.setPointSize(14)
        self.show2.setFont(font)
        self.show2.setAlignment(QtCore.Qt.AlignCenter)
        self.show2.setObjectName("show2")
        self.final3 = QtWidgets.QLineEdit(eightPuzzle)
        self.final3.setGeometry(QtCore.QRect(280, 60, 31, 31))
        self.final3.setAlignment(QtCore.Qt.AlignCenter)
        self.final3.setObjectName("final3")
        self.final4 = QtWidgets.QLineEdit(eightPuzzle)
        self.final4.setGeometry(QtCore.QRect(200, 100, 31, 31))
        self.final4.setAlignment(QtCore.Qt.AlignCenter)
        self.final4.setObjectName("final4")
        self.show7 = QtWidgets.QLabel(eightPuzzle)
        self.show7.setGeometry(QtCore.QRect(580, 150, 50, 50))
        font = QtGui.QFont()
        font.setFamily("Geeza Pro")
        font.setPointSize(14)
        self.show7.setFont(font)
        self.show7.setAlignment(QtCore.Qt.AlignCenter)
        self.show7.setObjectName("show7")
        self.final6 = QtWidgets.QLineEdit(eightPuzzle)
        self.final6.setGeometry(QtCore.QRect(280, 100, 31, 31))
        self.final6.setAlignment(QtCore.Qt.AlignCenter)
        self.final6.setObjectName("final6")
        self.show6 = QtWidgets.QLabel(eightPuzzle)
        self.show6.setGeometry(QtCore.QRect(680, 100, 50, 50))
        font = QtGui.QFont()
        font.setFamily("Geeza Pro")
        font.setPointSize(14)
        self.show6.setFont(font)
        self.show6.setAlignment(QtCore.Qt.AlignCenter)
        self.show6.setObjectName("show6")
        self.init1 = QtWidgets.QLineEdit(eightPuzzle)
        self.init1.setGeometry(QtCore.QRect(50, 60, 31, 31))
        self.init1.setAlignment(QtCore.Qt.AlignCenter)
        self.init1.setObjectName("init1")
        self.final1 = QtWidgets.QLineEdit(eightPuzzle)
        self.final1.setGeometry(QtCore.QRect(200, 60, 31, 31))
        self.final1.setAlignment(QtCore.Qt.AlignCenter)
        self.final1.setObjectName("final1")
        self.init8 = QtWidgets.QLineEdit(eightPuzzle)
        self.init8.setGeometry(QtCore.QRect(90, 140, 31, 31))
        self.init8.setAlignment(QtCore.Qt.AlignCenter)
        self.init8.setObjectName("init8")
        self.show8 = QtWidgets.QLabel(eightPuzzle)
        self.show8.setGeometry(QtCore.QRect(630, 150, 50, 50))
        font = QtGui.QFont()
        font.setFamily("Geeza Pro")
        font.setPointSize(14)
        self.show8.setFont(font)
        self.show8.setAlignment(QtCore.Qt.AlignCenter)
        self.show8.setObjectName("show8")
        self.show9 = QtWidgets.QLabel(eightPuzzle)
        self.show9.setGeometry(QtCore.QRect(680, 150, 50, 50))
        font = QtGui.QFont()
        font.setFamily("Geeza Pro")
        font.setPointSize(14)
        self.show9.setFont(font)
        self.show9.setText("")
        self.show9.setAlignment(QtCore.Qt.AlignCenter)
        self.show9.setObjectName("show9")
        self.show4 = QtWidgets.QLabel(eightPuzzle)
        self.show4.setGeometry(QtCore.QRect(580, 100, 50, 50))
        font = QtGui.QFont()
        font.setFamily("Geeza Pro")
        font.setPointSize(14)
        self.show4.setFont(font)
        self.show4.setAlignment(QtCore.Qt.AlignCenter)
        self.show4.setObjectName("show4")
        self.show3 = QtWidgets.QLabel(eightPuzzle)
        self.show3.setGeometry(QtCore.QRect(680, 50, 50, 50))
        font = QtGui.QFont()
        font.setFamily("Geeza Pro")
        font.setPointSize(14)
        self.show3.setFont(font)
        self.show3.setAlignment(QtCore.Qt.AlignCenter)
        self.show3.setObjectName("show3")
        self.Run = QtWidgets.QPushButton(eightPuzzle)
        self.Run.setGeometry(QtCore.QRect(50, 320, 71, 31))
        self.Run.setObjectName("Run")
        self.init2 = QtWidgets.QLineEdit(eightPuzzle)
        self.init2.setGeometry(QtCore.QRect(90, 60, 31, 31))
        self.init2.setAlignment(QtCore.Qt.AlignCenter)
        self.init2.setObjectName("init2")
        self.final8 = QtWidgets.QLineEdit(eightPuzzle)
        self.final8.setGeometry(QtCore.QRect(240, 140, 31, 31))
        self.final8.setAlignment(QtCore.Qt.AlignCenter)
        self.final8.setObjectName("final8")
        self.final5 = QtWidgets.QLineEdit(eightPuzzle)
        self.final5.setGeometry(QtCore.QRect(240, 100, 31, 31))
        self.final5.setAlignment(QtCore.Qt.AlignCenter)
        self.final5.setObjectName("final5")
        self.init4 = QtWidgets.QLineEdit(eightPuzzle)
        self.init4.setGeometry(QtCore.QRect(50, 100, 31, 31))
        self.init4.setAlignment(QtCore.Qt.AlignCenter)
        self.init4.setObjectName("init4")
        self.label_9 = QtWidgets.QLabel(eightPuzzle)
        self.label_9.setGeometry(QtCore.QRect(70, 10, 71, 31))
        self.label_9.setAlignment(QtCore.Qt.AlignCenter)
        self.label_9.setObjectName("label_9")
        self.Log = QtWidgets.QPushButton(eightPuzzle)
        self.Log.setGeometry(QtCore.QRect(830, 150, 71, 31))
        self.Log.setObjectName("Log")
        self.medline = QtWidgets.QFrame(eightPuzzle)
        self.medline.setGeometry(QtCore.QRect(455, 0, 16, 601))
        self.medline.setLineWidth(1)
        self.medline.setFrameShape(QtWidgets.QFrame.VLine)
        self.medline.setFrameShadow(QtWidgets.QFrame.Sunken)
        self.medline.setObjectName("medline")
        self.Algorithm_lable = QtWidgets.QLabel(eightPuzzle)
        self.Algorithm_lable.setGeometry(QtCore.QRect(50, 220, 131, 16))
        self.Algorithm_lable.setObjectName("Algorithm_lable")
        self.algorithm_comboBox = QtWidgets.QComboBox(eightPuzzle)
        self.algorithm_comboBox.setGeometry(QtCore.QRect(50, 250, 75, 23))
        self.algorithm_comboBox.setEditable(False)
        self.algorithm_comboBox.setSizeAdjustPolicy(QtWidgets.QComboBox.AdjustToContentsOnFirstShow)
        self.algorithm_comboBox.setObjectName("algorithm_comboBox")
        self.algorithm_comboBox.addItem("")
        self.algorithm_comboBox.addItem("")
        self.algorithm_comboBox.addItem("")
        self.process_label = QtWidgets.QLabel(eightPuzzle)
        self.process_label.setGeometry(QtCore.QRect(490, 20, 54, 12))
        self.process_label.setObjectName("process_label")
        self.Restore = QtWidgets.QPushButton(eightPuzzle)
        self.Restore.setGeometry(QtCore.QRect(160, 320, 71, 31))
        self.Restore.setObjectName("Restore")
        self.line = QtWidgets.QFrame(eightPuzzle)
        self.line.setGeometry(QtCore.QRect(460, 190, 481, 20))
        self.line.setFrameShape(QtWidgets.QFrame.HLine)
        self.line.setFrameShadow(QtWidgets.QFrame.Sunken)
        self.line.setObjectName("line")
        self.label = QtWidgets.QLabel(eightPuzzle)
        self.label.setGeometry(QtCore.QRect(490, 260, 81, 16))
        self.label.setObjectName("label")
        self.lineEdit = QtWidgets.QLineEdit(eightPuzzle)
        self.lineEdit.setGeometry(QtCore.QRect(580, 260, 41, 21))
        self.lineEdit.setObjectName("lineEdit")
        self.Go = QtWidgets.QPushButton(eightPuzzle)
        self.Go.setGeometry(QtCore.QRect(550, 320, 71, 31))
        self.Go.setObjectName("Go")
        self.show1_2 = QtWidgets.QLabel(eightPuzzle)
        self.show1_2.setGeometry(QtCore.QRect(710, 230, 50, 50))
        font = QtGui.QFont()
        font.setFamily("Geeza Pro")
        font.setPointSize(14)
        self.show1_2.setFont(font)
        self.show1_2.setLayoutDirection(QtCore.Qt.LeftToRight)
        self.show1_2.setAlignment(QtCore.Qt.AlignCenter)
        self.show1_2.setObjectName("show1_2")
        self.show5_2 = QtWidgets.QLabel(eightPuzzle)
        self.show5_2.setGeometry(QtCore.QRect(760, 280, 50, 50))
        font = QtGui.QFont()
        font.setFamily("Geeza Pro")
        font.setPointSize(14)
        self.show5_2.setFont(font)
        self.show5_2.setAlignment(QtCore.Qt.AlignCenter)
        self.show5_2.setObjectName("show5_2")
        self.show7_2 = QtWidgets.QLabel(eightPuzzle)
        self.show7_2.setGeometry(QtCore.QRect(710, 330, 50, 50))
        font = QtGui.QFont()
        font.setFamily("Geeza Pro")
        font.setPointSize(14)
        self.show7_2.setFont(font)
        self.show7_2.setAlignment(QtCore.Qt.AlignCenter)
        self.show7_2.setObjectName("show7_2")
        self.show6_2 = QtWidgets.QLabel(eightPuzzle)
        self.show6_2.setGeometry(QtCore.QRect(810, 280, 50, 50))
        font = QtGui.QFont()
        font.setFamily("Geeza Pro")
        font.setPointSize(14)
        self.show6_2.setFont(font)
        self.show6_2.setAlignment(QtCore.Qt.AlignCenter)
        self.show6_2.setObjectName("show6_2")
        self.show8_2 = QtWidgets.QLabel(eightPuzzle)
        self.show8_2.setGeometry(QtCore.QRect(760, 330, 50, 50))
        font = QtGui.QFont()
        font.setFamily("Geeza Pro")
        font.setPointSize(14)
        self.show8_2.setFont(font)
        self.show8_2.setAlignment(QtCore.Qt.AlignCenter)
        self.show8_2.setObjectName("show8_2")
        self.show9_2 = QtWidgets.QLabel(eightPuzzle)
        self.show9_2.setGeometry(QtCore.QRect(810, 330, 50, 50))
        font = QtGui.QFont()
        font.setFamily("Geeza Pro")
        font.setPointSize(14)
        self.show9_2.setFont(font)
        self.show9_2.setText("")
        self.show9_2.setAlignment(QtCore.Qt.AlignCenter)
        self.show9_2.setObjectName("show9_2")
        self.show3_2 = QtWidgets.QLabel(eightPuzzle)
        self.show3_2.setGeometry(QtCore.QRect(810, 230, 50, 50))
        font = QtGui.QFont()
        font.setFamily("Geeza Pro")
        font.setPointSize(14)
        self.show3_2.setFont(font)
        self.show3_2.setAlignment(QtCore.Qt.AlignCenter)
        self.show3_2.setObjectName("show3_2")
        self.show2_2 = QtWidgets.QLabel(eightPuzzle)
        self.show2_2.setGeometry(QtCore.QRect(760, 230, 50, 50))
        font = QtGui.QFont()
        font.setFamily("Geeza Pro")
        font.setPointSize(14)
        self.show2_2.setFont(font)
        self.show2_2.setAlignment(QtCore.Qt.AlignCenter)
        self.show2_2.setObjectName("show2_2")
        self.show4_2 = QtWidgets.QLabel(eightPuzzle)
        self.show4_2.setGeometry(QtCore.QRect(710, 280, 50, 50))
        font = QtGui.QFont()
        font.setFamily("Geeza Pro")
        font.setPointSize(14)
        self.show4_2.setFont(font)
        self.show4_2.setAlignment(QtCore.Qt.AlignCenter)
        self.show4_2.setObjectName("show4_2")

        self.retranslateUi(eightPuzzle)
        QtCore.QMetaObject.connectSlotsByName(eightPuzzle)

    def retranslateUi(self, eightPuzzle):
        _translate = QtCore.QCoreApplication.translate
        eightPuzzle.setWindowTitle(_translate("eightPuzzle", "eightPuzzle"))
        self.show1.setText(_translate("eightPuzzle", "-"))
        self.show5.setText(_translate("eightPuzzle", "-"))
        self.label_10.setText(_translate("eightPuzzle", "目标状态"))
        self.show2.setText(_translate("eightPuzzle", "-"))
        self.show7.setText(_translate("eightPuzzle", "-"))
        self.show6.setText(_translate("eightPuzzle", "-"))
        self.show8.setText(_translate("eightPuzzle", "-"))
        self.show4.setText(_translate("eightPuzzle", "-"))
        self.show3.setText(_translate("eightPuzzle", "-"))
        self.Run.setText(_translate("eightPuzzle", "Run"))
        self.label_9.setText(_translate("eightPuzzle", "初始状态"))
        self.Log.setText(_translate("eightPuzzle", "log"))
        self.Algorithm_lable.setText(_translate("eightPuzzle", "Choose the algorithm"))
        self.algorithm_comboBox.setItemText(0, _translate("eightPuzzle", "A"))
        self.algorithm_comboBox.setItemText(1, _translate("eightPuzzle", "宽度搜索"))
        self.algorithm_comboBox.setItemText(2, _translate("eightPuzzle", "深度搜索"))
        self.process_label.setText(_translate("eightPuzzle", "process"))
        self.Restore.setText(_translate("eightPuzzle", "Restore"))
        self.label.setText(_translate("eightPuzzle", "查看具体步骤"))
        self.Go.setText(_translate("eightPuzzle", "Go"))
        self.show1_2.setText(_translate("eightPuzzle", "-"))
        self.show5_2.setText(_translate("eightPuzzle", "-"))
        self.show7_2.setText(_translate("eightPuzzle", "-"))
        self.show6_2.setText(_translate("eightPuzzle", "-"))
        self.show8_2.setText(_translate("eightPuzzle", "-"))
        self.show3_2.setText(_translate("eightPuzzle", "-"))
        self.show2_2.setText(_translate("eightPuzzle", "-"))
        self.show4_2.setText(_translate("eightPuzzle", "-"))
